### PR TITLE
Add CI to release App

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
This is the first iteration of the circleci config that takes care of publishing the App bundle into S3. I've also added some basic test, build & lint CI.

`make test` fails because there are no tests.
`make build` fails because of a lot of ts issues.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-33649

### CODE REVIEW

**Review by iomodo — DISMISSED** [review]
I assume fixes will be the part of the different PR

File: README.md [hunk 1]
```diff
@@ -31,9 +31,8 @@ mm-plugin-apps: `master` @831d203
             1. Ex. `https://https://testing.ngrok.io`
     1. `ZD_CLIENT_SECRET` - (will be set later in the setup)
 1. Start the node server
-    1. `npm i` - install node_moduls and dependencies
-    1. `npm run build:watch` - (to monitor typescriopt errors and watch chaning files errors)
-    1. `npm start` - (in a separate shell) start the node server
+    1. `make watch` - (to monitor typescriopt errors and watch chaning files errors)
```
Comment (jfrerich) [inline README.md:34]:
```suggestion
    1. `make watch` - (to monitor typescript errors and watch changing files errors)
```

**Review by jfrerich — DISMISSED** [review]
Thanks @hanzei!  I do like this but have not tested it.  I can always workaround any issues that show up

File: README.md (hunk 1 above)
Comment (jfrerich) [reply README.md]: I fixed this with latest merge to master conflict.

**Review by jfrerich — APPROVED** [review]
@hanzei, I just merged master, fixed the spelling changes, and tested `make run` and `make watch`.

**Pull request merged**